var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -255,6 +257,9 @@ def configure_args(self):
             extra_args.append('--disable-shared')
             extra_args.append('--enable-static-exec')
 
+        if '+static' in self.spec:
+            extra_args.append('--enable-static')
```

Comment:
Can you explicitly disable this in an else clause?

Reply:
Done! I also made the config line more explicit by adding
```
--enable-static=[=PKGS]  build static libraries [default=yes]
```
which is more in line with how hdf5's `./configure --help` has it structured. (Thanks for pointing that out @alalazo)